diff -ru dash-0.5.7/src/bltin/test.c dash-0.5.7-new/src/bltin/test.c
--- dash-0.5.7/src/bltin/test.c	2011-03-15 00:18:06.000000000 -0700
+++ dash-0.5.7-new/src/bltin/test.c	2017-06-06 12:57:18.211291543 -0700
@@ -528,9 +528,9 @@
 	if (gid == getgid() || gid == getegid())
 		return (1);
 
-	ngroups = getgroups(0, NULL);
-	group_array = stalloc(ngroups * sizeof(gid_t));
-	if ((getgroups(ngroups, group_array)) != ngroups)
+	//ngroups = getgroups(0, NULL);
+	//group_array = stalloc(ngroups * sizeof(gid_t));
+	//if ((getgroups(ngroups, group_array)) != ngroups)
 		return (0);
 
 	/* Search through the list looking for GID. */
diff -ru dash-0.5.7/src/builtins.def.in dash-0.5.7-new/src/builtins.def.in
--- dash-0.5.7/src/builtins.def.in	2010-05-26 20:45:23.000000000 -0700
+++ dash-0.5.7-new/src/builtins.def.in	2017-06-06 12:58:52.892326015 -0700
@@ -46,7 +46,7 @@
  */
 
 #ifndef JOBS
-#define JOBS 1
+#define JOBS 0
 #endif
 
 #if JOBS
diff -ru dash-0.5.7/src/jobs.c dash-0.5.7-new/src/jobs.c
--- dash-0.5.7/src/jobs.c	2011-03-15 00:45:32.000000000 -0700
+++ dash-0.5.7-new/src/jobs.c	2017-06-06 13:03:15.904978131 -0700
@@ -46,7 +46,7 @@
 #include <sys/time.h>
 #include <sys/resource.h>
 #endif
-#include <sys/ioctl.h>
+//#include <sys/ioctl.h>
 
 #include "shell.h"
 #if JOBS
@@ -1136,17 +1136,17 @@
 
 	do {
 		gotsigchld = 0;
-		err = wait3(status, flags, NULL);
+		err = waitpid(-1, status, flags);
 		if (err || !block)
 			break;
 
 		block = 0;
 
 		sigfillset(&mask);
-		sigprocmask(SIG_SETMASK, &mask, &oldmask);
+		//sigprocmask(SIG_SETMASK, &mask, &oldmask);
 
-		while (!gotsigchld && !pendingsigs)
-			sigsuspend(&oldmask);
+		//while (!gotsigchld && !pendingsigs)
+		//	sigsuspend(&oldmask);
 
 		sigclearmask();
 	} while (gotsigchld);
diff -ru dash-0.5.7/src/mkinit.c dash-0.5.7-new/src/mkinit.c
--- dash-0.5.7/src/mkinit.c	2008-05-01 23:09:44.000000000 -0700
+++ dash-0.5.7-new/src/mkinit.c	2017-06-06 12:51:08.200809714 -0700
@@ -47,7 +47,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <fcntl.h>
+//#include <fcntl.h>
 #include <unistd.h>
 
 
diff -ru dash-0.5.7/src/output.c dash-0.5.7-new/src/output.c
--- dash-0.5.7/src/output.c	2011-03-15 00:18:06.000000000 -0700
+++ dash-0.5.7-new/src/output.c	2017-06-06 12:18:38.123364338 -0700
@@ -45,7 +45,7 @@
 
 #include <sys/types.h>		/* quad_t */
 #include <sys/param.h>		/* BSD4_4 */
-#include <sys/ioctl.h>
+//#include <sys/ioctl.h>
 
 #include <stdio.h>	/* defines BUFSIZ */
 #include <string.h>
diff -ru dash-0.5.7/src/shell.h dash-0.5.7-new/src/shell.h
--- dash-0.5.7/src/shell.h	2008-05-01 23:09:44.000000000 -0700
+++ dash-0.5.7-new/src/shell.h	2017-06-06 12:33:14.200424324 -0700
@@ -52,7 +52,7 @@
 #include <sys/param.h>
 
 #ifndef JOBS
-#define JOBS 1
+#define JOBS 0
 #endif
 #ifndef BSD
 #define BSD 1
diff -ru dash-0.5.7/src/system.h dash-0.5.7-new/src/system.h
--- dash-0.5.7/src/system.h	2011-03-15 00:34:53.000000000 -0700
+++ dash-0.5.7-new/src/system.h	2017-06-06 12:54:36.082716644 -0700
@@ -36,9 +36,8 @@
 
 static inline void sigclearmask(void)
 {
-#ifdef HAVE_SIGSETMASK
+#if 0
 	sigsetmask(0);
-#else
 	sigset_t set;
 	sigemptyset(&set);
 	sigprocmask(SIG_SETMASK, &set, 0);
@@ -82,7 +81,7 @@
 	      int (*)(const void *, const void *));
 #endif
 
-#ifndef HAVE_KILLPG
+#if 0
 static inline int killpg(pid_t pid, int signal)
 {
 #ifdef DEBUG
diff -ru dash-0.5.7/src/trap.c dash-0.5.7-new/src/trap.c
--- dash-0.5.7/src/trap.c	2010-11-28 04:54:10.000000000 -0800
+++ dash-0.5.7-new/src/trap.c	2017-06-06 12:55:47.773584397 -0700
@@ -183,6 +183,7 @@
 void
 setsignal(int signo)
 {
+#if 0
 	int action;
 	char *t, tsig;
 	struct sigaction act;
@@ -262,6 +263,7 @@
 	act.sa_flags = 0;
 	sigfillset(&act.sa_mask);
 	sigaction(signo, &act, 0);
+#endif
 }
 
 /*
diff -ru dash-0.5.7/src/var.c dash-0.5.7-new/src/var.c
--- dash-0.5.7/src/var.c	2011-03-15 00:45:32.000000000 -0700
+++ dash-0.5.7-new/src/var.c	2017-06-06 13:04:12.695518184 -0700
@@ -139,7 +139,7 @@
 		}
 	}
 
-	fmtstr(ppid + 5, sizeof(ppid) - 5, "%ld", (long) getppid());
+	fmtstr(ppid + 5, sizeof(ppid) - 5, "%ld", 1);
 	setvareq(ppid, VTEXTFIXED);
 
 	p = lookupvar("PWD");
