diff -rupN dosbox/config.sub dosbox-redox/config.sub
--- dosbox/config.sub	2010-05-10 10:43:54.000000000 -0700
+++ dosbox-redox/config.sub	2016-07-27 18:05:56.382451841 -0700
@@ -1288,7 +1288,7 @@ case $os in
 	-gnu* | -bsd* | -mach* | -minix* | -genix* | -ultrix* | -irix* \
 	      | -*vms* | -sco* | -esix* | -isc* | -aix* | -cnk* | -sunos | -sunos[34]*\
 	      | -hpux* | -unos* | -osf* | -luna* | -dgux* | -auroraux* | -solaris* \
-	      | -sym* | -kopensolaris* \
+	      | -sym* | -kopensolaris* | -redox* \
 	      | -amigaos* | -amigados* | -msdos* | -newsos* | -unicos* | -aof* \
 	      | -aos* | -aros* \
 	      | -nindy* | -vxsim* | -vxworks* | -ebmon* | -hms* | -mvs* \
diff -rupN dosbox/include/cross.h dosbox-redox/include/cross.h
--- dosbox/include/cross.h	2010-05-10 10:43:54.000000000 -0700
+++ dosbox-redox/include/cross.h	2016-07-27 18:05:56.382451841 -0700
@@ -61,6 +61,7 @@
 #define ftruncate(blah,blah2) chsize(blah,blah2)
 #endif
 
+#undef DB_HAVE_NO_POWF
 //Solaris maybe others
 #if defined (DB_HAVE_NO_POWF)
 #include <math.h>
diff -rupN dosbox/src/cpu/cpu.cpp dosbox-redox/src/cpu/cpu.cpp
--- dosbox/src/cpu/cpu.cpp	2010-05-12 02:57:31.000000000 -0700
+++ dosbox-redox/src/cpu/cpu.cpp	2016-07-27 18:05:56.382451841 -0700
@@ -22,7 +22,7 @@
 #include <sstream>
 #include "dosbox.h"
 #include "cpu.h"
-#include "memory.h"
+//#include "memory.h"
 #include "debug.h"
 #include "mapper.h"
 #include "setup.h"
diff -rupN dosbox/src/dos/cdrom_image.cpp dosbox-redox/src/dos/cdrom_image.cpp
--- dosbox/src/dos/cdrom_image.cpp	2010-05-10 10:43:54.000000000 -0700
+++ dosbox-redox/src/dos/cdrom_image.cpp	2016-07-27 18:05:56.382451841 -0700
@@ -418,6 +418,8 @@ static string dirname(char * file) {
 
 bool CDROM_Interface_Image::LoadCueSheet(char *cuefile)
 {
+	return false;
+	/*
 	Track track = {0, 0, 0, 0, 0, 0, false, NULL};
 	tracks.clear();
 	int shift = 0;
@@ -549,6 +551,7 @@ bool CDROM_Interface_Image::LoadCueSheet
 	if(!AddTrack(track, shift, 0, totalPregap, 0)) return false;
 
 	return true;
+	*/
 }
 
 bool CDROM_Interface_Image::AddTrack(Track &curr, int &shift, int prestart, int &totalPregap, int currPregap)
diff -rupN dosbox/src/misc/cross.cpp dosbox-redox/src/misc/cross.cpp
--- dosbox/src/misc/cross.cpp	2010-05-10 10:43:54.000000000 -0700
+++ dosbox-redox/src/misc/cross.cpp	2016-07-27 18:05:56.382451841 -0700
@@ -103,6 +103,7 @@ void Cross::ResolveHomedir(std::string &
 	if(temp_line.size() == 1 || temp_line[1] == CROSS_FILESPLIT) { //The ~ and ~/ variant
 		char * home = getenv("HOME");
 		if(home) temp_line.replace(0,1,std::string(home));
+/*
 #if defined HAVE_SYS_TYPES_H && defined HAVE_PWD_H
 	} else { // The ~username variant
 		std::string::size_type namelen = temp_line.find(CROSS_FILESPLIT);
@@ -111,6 +112,7 @@ void Cross::ResolveHomedir(std::string &
 		struct passwd* pass = getpwnam(username.c_str());
 		if(pass) temp_line.replace(0,namelen,pass->pw_dir); //namelen -1 +1(for the ~)
 #endif // USERNAME lookup code
+*/
 	}
 }
 
